test.py

```diff
@@ -356,6 +356,7 @@ def __init__(self, path, cfg: dict, options: argparse.Namespace, mode: str) -> N
         cluster_cfg = self.cfg.get("cluster", {"initial_size": 1})
         cluster_size = cluster_cfg["initial_size"]
         pool_size = cfg.get("pool_size", 2)
+        self.exclusive_cluster = set(cfg.get("exclusive_cluster", []))
```

Comment:
nitpick: I'm not sure the phrase word "exclusive cluster" is self-explanatory. Especially since it doesn't even mean it gets a new exclusive cluster - but rather that it could get a "recycled" cluster but it will be the last one to use it.
Maybe the name "dirties_cluster" is better - it's a test which dirties the cluster (with the word "dirty" being the terminology already used by test.py).